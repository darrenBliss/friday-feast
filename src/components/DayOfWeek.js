// @flow
import * as React from 'react'

type Props = {
  dayIndex: number
}

const DayOfWeek = ({ dayIndex }: Props) => {
  if (dayIndex < 1 || dayIndex > 7 ) {
    return
  }
  const today: Date = new Date()
  const theDay: Date = new Date(today.getFullYear(), today.getMonth(), today.getDate() - today.getDay() + dayIndex)

  return (
    <React.Fragment>{`
      ${theDay.toLocaleString('en-GB', { weekday: "long" })}
      ${theDay.getDate()}
      ${theDay.toLocaleString('en-GB', { month: "long" })}
      ${theDay.getFullYear()}
    `}</React.Fragment>
  )
}

DayOfWeek.defaultProps = {
  dayIndex: 5
}

export default DayOfWeek