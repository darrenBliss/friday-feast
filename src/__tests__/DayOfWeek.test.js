import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import DayOfWeek from '../components/DayOfWeek'

describe('Date component should render', () => {
  it('test enzyme snapshot works', () => {
    const wrapper = shallow(<DayOfWeek />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('test enzyme renders Tuesday in snapshot', () => {
    const wrapper = shallow(<DayOfWeek dayIndex={2} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
