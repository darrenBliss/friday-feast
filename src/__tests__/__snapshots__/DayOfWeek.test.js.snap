// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date component Render Friday as default 1`] = `
<React.Fragment>
  
      Friday
      25
      May
      2018
    
</React.Fragment>
`;

exports[`Date component Renders Tuesday in snapshot 1`] = `
<React.Fragment>
  
      Tuesday
      22
      May
      2018
    
</React.Fragment>
`;

exports[`Date component test number out of week 1`] = `""`;
